src/framework/components/anim/controller.js

```diff
@@ -182,7 +182,7 @@ class BlendTree extends AnimNode {
                 parameterValues = this._parameters.map(function (param) {
                     return this._findParameter(param).value;
                 }.bind(this));
-                if (this._parameterValues && (this._parameterValues.equals(parameterValues))) {
+                if (this._parameterValues && (JSON.stringify(this._parameterValues) === JSON.stringify(parameterValues))) {
```

Comment:
This is a very slow way to deeply compare data and leads to allocations. If the structure of the provided values known, then it is better to write a specific comparison method, which fails early if deviations are found.

Reply:
Good catch here. I've addressed this in another PR which will go in before this one.